projects/gateway2/translator/sslutils/ssl_utils.go

```
@@ -45,5 +47,20 @@ func isValidSslKeyPair(certChain, privateKey, rootCa []byte) error {
 	}
 
 	_, err := tls.X509KeyPair([]byte(certChain), []byte(privateKey))
+	if err != nil {
+		return err
+	}
+	candidateCert, err := cert.ParseCertsPEM([]byte(certChain))
+	if err != nil {
+		// return err rather than sanitize. This is to maintain UX with older versions and to prevent a large refactor
+		// for gatewayv2 secret validation code.
+		return err
+	}
+	reencoded, err := cert.EncodeCertificates(candidateCert...)
```

We'll eventually return this error if it's non-nil, but first we return our own error if the cert doesn't match expectations

I would expect that we should actually return the error right away, but if not can we justify the behavior with a comment